app/code/Magento/ConfigurableProduct/Model/ResourceModel/Product/Type/Configurable.php

```diff
@@ -46,6 +43,13 @@ class Configurable extends \Magento\Framework\Model\ResourceModel\Db\AbstractDb
      */
     private $optionProvider;
 
+    /**
+     * List of attribute options data
+     *
+     * @var array
+     */
+    protected $attributeOptions = [];
```

Comment:
This should be marked as `private` instead of `protected` right?
If classes extend from this one (which is against Magento's policy) they'll need to call the public function to get the value, not use this property.
There is no good reason (unless I'm missing something) to expose this property to classes that extend from this one, right?